src/common/serac_types.hpp

```diff
@@ -54,11 +56,11 @@ struct LinearSolverParameters {
 
 // Finite element information bundle
 struct FiniteElementState {
-  mfem::ParFiniteElementSpace* space = nullptr;
-  mfem::FiniteElementCollection* coll = nullptr;
-  mfem::ParGridFunction* gf = nullptr;
-  mfem::ParMesh* mesh = nullptr;
-  mfem::Vector* true_vec = nullptr;
+  std::shared_ptr < mfem::ParFiniteElementSpace > space;
```

Comment:
I feel like space, coll, and mesh should be getters to gf. Otherwise there could be a danger that space, coll, mesh, gf could be inconsistent.

Reply:
With MFEM I believe you always have to make sure the space, coll, mesh are in scope, both with raw and smart pointers otherwise the gridfunction doesn't really make sense. Maybe what you are getting at is how to best guarantee the existence of space, coll, mesh for a given rid function. I think `FiniteElementState` could take in a shared_ptrs and then construct a ParGridFunction for you. I forget what happens in the case of NCMeshes.

I still think this should be abstracted into a class. If it's not in this PR, we should put it in a separate one. There's a big benefit to using Getters to access Space() and so forth as we can change the underlying guts without messing with the rest of the code. This would allow us to change to enums or under the hood seamlessly.